alephium/contracts/governance.ral

```diff
@@ -0,0 +1,202 @@
+Contract Governance(
+    chainId: U256,
+    governanceChainId: U256,
+    governanceEmitterAddress: ByteVec,
+    mut receivedSequence: U256,
+    mut messageFee: U256,
+    mut guardianSets: [ByteVec; 2], // keep previous and current guardian set
+    mut guardianSetIndexes: [U256; 2],
+    mut previousGuardianSetExpirationTime: U256
+) extends Constants() {
+    event WormholeMessage(sender: ByteVec, targetChainId: U256, sequence: U256, nonce: ByteVec, payload: ByteVec, consistencyLevel: U256)
+
+    const Version = #01
+    const GuardianSetExpireDuration = 86400000 // one day in ms
+    const CoreModule = 0x436f7265 // module identifier: "Core"
+
+    enum ActionId {
+        ContractUpgrade = #01
+        NewGuardianSet  = #02
+        NewMessageFee   = #03
+        TransferFee     = #04
+    }
+
+    @using(readonly = true, permissionCheck = false)
```

Comment:
Do we need to have `permissionCheck = false` here?

Reply:
Yes, `permissionCheck` is true for public functions by default, we need to make it false to avoid compiler warnings.